tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12930,6 +12930,41 @@ bool ValidateAttributeTargetIsFunction(Sema& S, Decl* D, const AttributeList &A)
   return false;
 }
 
+HLSLShaderAttr* ValidateShaderAttributes(Sema &S, Decl *D,
+                                         const AttributeList &A) {
+  StringRef sRef = ValidateAttributeStringArg(
```

Comment:
Is there a reason to use this old validation code instead of just handling this below when we call `KindFromFullName`?